src/Agent.Listener/SelfUpdater.cs

```diff
@@ -176,7 +176,19 @@ private async Task<bool> HashValidation(string archiveFile)
                 return true;
             }
 
-            bool isHostedServer = await _serverUtil.IsDeploymentTypeHosted(_serverUrl, _creds, _locationServer);
+            bool isHostedServer;
+            try
+            {
+                isHostedServer = await _serverUtil.IsDeploymentTypeHosted(_serverUrl, _creds, _locationServer);
+            }
+            catch (DeploymentTypeNotDeterminedException ex)
+            {
+                throw new DeploymentTypeNotDeterminedException(ex.Message + $@"
+This exception was thrown during checksum validation when performing the agent self-update process.
```

Comment:
Will this be always an exception for the customers?

Reply:
Yes, there should always be an exception thrown for the customers in this case.
The customers can avoid this by setting the `DISABLE_HASH_VALIDATION` environment variable to `true`.